### ISSUE TITLE
[Feature Request] Provide a repo

### ISSUE THREAD

**Issue by DC7IA** [loki-project/session-desktop#1096]
I would like to suggest providing a repo for easier updates.

Downloading packages by hand is soooo much work. :D

**Comment by hueq** [comment]
Making repositories available would be more convenient for users, increasing the chance their software is kept up to date. Currently, each new version must be downloaded and verified by hand.

However, what is the security of repositories like? My understanding is APT (in Debian) by default only signs/verifies the repository data, but downloaded packages are not signed/verified. But I might be wrong. Is there a way for Loki to sign packages as well as repository data?

About software update anonymity, Debian has onion addresses for their repositories, so that it's difficult for an attacker to know a user is performing software updates (at the user's end) or who is performing software updates (at the repository end), therefore difficult for an attacker to feed targeted malware. Obviously it would be good to see this done over Lokinet too.

I'm not sure what to say about Windows and Mac.

**Comment by jagerman** [comment]
> However, what is the security of repositories like? My understanding is APT (in Debian) by default only signs/verifies the repository data, but downloaded packages are not signed/verified.

This is not correct; there is indeed only one signed file, `Release`, but this file includes SHA256 checksums of every `Packages` files for each distribution and architecture; the Packages files in turn contain SHA256 checksums of each individual deb.  So while there is only one signature, that signature applies transitively to every file in the repository.

**Comment by hueq** [comment]
> This is not correct; there is indeed only one signed file, `Release`, but this file includes SHA256 checksums of every `Packages` files for each distribution and architecture; the Packages files in turn contain SHA256 checksums of each individual deb. So while there is only one signature, that signature applies transitively to every file in the repository.

I believe what you say is true about the `Releases` -> `Packages` -> package verification chain. So as long as the signature and all the checksums are properly checked as being OK (and signed with the correct key) there is an assurance of good download integrity.

I wonder if there are weaknesses in this security model though, like what would happen if a high-profile dev (whose signing key is in the Debian keyring for example) signs a malicious `Release` file of a repository they don't own? This appears to be mitigated by specifying the signing key explicitly, and some software projects instruct people to add something like this to their repo list:

~~~
deb [signed-by=/usr/share/keyrings/name-of-keyring.gpg] https://example.org/debian/ xenial main
~~~

APT security is a topic for APT not Session so I don't want to go deeper, but I mention it here to inform the decision about this issue. I'd be happy to see Session available as a repo though.

**Comment by jagerman** [comment]
Packages are available in the Loki apt repo - see #1397 for details.
